app/Compile.hs

```diff
@@ -43,32 +36,47 @@ typecheck fin Michelson = do
   ctx <- parse fin
   let res = Pipeline.contextToCore ctx Param.michelson
   case res of
-    Right (FF.CoreDefs order globals) -> do
-      let globalDefs = HM.mapMaybe (\case (CoreDef g) -> pure g; _ -> Nothing) globals
-      case HM.elems $ HM.filter (\x -> case x of (IR.GFunction (IR.Function (_ :| ["main"]) _ _ _)) -> True; _ -> False) globalDefs of
+    Right (FF.CoreDefs _order globals) -> do
+      let globalDefs = HM.mapMaybe toCoreDef globals
+      case HM.elems $ HM.filter isMain globalDefs of
         [] -> do
           T.putStrLn "No main function found"
           exitFailure
-        func@(IR.GFunction (IR.Function name usage ty (IR.FunClause _ [] term _ _ _ :| []))) : [] -> do
-          let newGlobals = HM.map (unsafeEvalGlobal (map convGlobal globalDefs)) globalDefs
-              inlinedTerm = IR.inlineAllGlobals (IR.injectT term) (\(IR.Global n) -> HM.lookup n globalDefs)
-          (res, _) <- exec (CorePipeline.coreToAnn (IR.extForgetT @(Juvix.Core.IR.TransformExt.OnlyExts.T IR.NoExt) inlinedTerm) (IR.globalToUsage usage) ty) Param.michelson newGlobals
-          case res of
-            Right r -> do
-              pure r
-            Left err -> do
-              print term
-              T.putStrLn (show err)
-              exitFailure
+        [IR.RawGFunction f]
```

Comment:
This should be cleaned up by @fdedden's work on a compositional pipeline, but it's fine in this PR for now